### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by gudrutis**
New PR.

@smuzaffar , can you give me some tips on how to proceed in checkOptionsLocalSource() function?
If I have to guess, I should call linux commands and execute it to given paths, something like:
```python
    if filename.endswith(".tar"):
        os.system("tar -cvf %s/%s %s" % (downloadDir, filename, localDir) )
        # should look like  os.system("tar -cvf tar2/tar1.tar tar1/ )

```

### CODE REVIEW

**Comment by smuzaffar** [comment]
@gudrutis , you should use getstatusoutput e.g.
```
error, output = getstatusoutput("tar -cvf %s/%s %s" % (downloadDir, filename, localDir))
```

File: cmsBuild [hunk 1]
```diff
@@ -634,21 +633,52 @@ class DownloadOptions(object):
...
+    # TODO-zm what todo for Source and Source0 being equal
+    if (not pkg.name in local_source_dict) or (not source_name in local_source_dict[pkg.name]):
         return
+    log("Package %s source %s is taken from local path  %s" % pkg.name, source_name)
 
-    pdb.set_trace()  #TODO-zm revmove
+    # TODO-zm
     """
-    TODO If pkg_name pkg_source in options, try to download from path
+    .rpm 4
```
Comment (smuzaffar) [inline cmsBuild]: we should fail if someone wants to override this type of file.

File: cmsBuild [hunk 2]
```diff
@@ -634,21 +633,52 @@ class DownloadOptions(object):
...
+    none 20
```
Comment (smuzaffar) [inline cmsBuild]: should be also be ignored. One can not override `none` source

**Comment by gudrutis** [comment]
@smuzaffar 

I was testing updated cmsBuild with cms-data packages, for example `data-L1Trigger-L1THGCal`, and I noticed that our solution is not sufficient - there is no source variable in the recipe.

https://github.com/cms-sw/cmsdist/blob/IB/CMSSW_10_3_X/gcc700/data-build-github.file
https://github.com/cms-sw/cmsdist/blob/IB/CMSSW_10_3_X/gcc700/data-L1Trigger-L1THGCal.spec

```bash
[#@cmsdev14 pkgtools_develop]$ ./pkgtools/cmsBuild -c cmsdist/ -a slc6_amd64_gcc630 -i temp_build/ -j 8 --repository=cms.week0 --sources build data-L1Trigger-L1THGCal
Read from cache: external+gcc+6.3.0-cms2
Generated cache: cms+data-L1Trigger-L1THGCal+V01-00-08
Package cms+data-L1Trigger-L1THGCal+V01-00-08 requested.
Sources for data-L1Trigger-L1THGCal are :
[#@cmsdev14 pkgtools_develop]$
```

File: cmsBuild [hunk 3]
```diff
@@ -636,19 +636,8 @@ class DownloadOptions(object):
...
-    # TODO-zm
-    def parseOptions (list):
-        return
-
-    pdb.set_trace()  #TODO-zm revmove
-    """
-    TODO If pkg_name pkg_source in options, try to download from path
-    """
+    source_name = pkg.sourcesNumbers[source] if pkg else None
+    pdb.set_trace()  #TODO-zm removex
```
Comment (smuzaffar) [inline cmsBuild]: haven't you removed pdb import?

File: cmsBuild [hunk 4]
```diff
@@ -637,42 +637,31 @@ def getLocalSources(pkg, source_name, downloadDir, filename):
...
-        return
-    if filename.endswith(".tar.xz"):
+    local_source_path = local_source_dict[pkg.name][source_name]
+    if filename.endswith(".rpm") or line.endswith("none"):
+        log("WARNING: the filename extension is not overwritable: %s " % filename)
         return
+    elif os.path.isfile(local_source_path):
+        error, output = getstatusoutput("cp -f %s %s/%s" % (local_source_path, downloadDir, filename))
+    elif filename.endswith(".tgz") or filename.endswith(".tar.gz"):
+        error, output = getstatusoutput("tar -cvf %s/%s %s" % (downloadDir, filename, local_source_path))
```
Comment (smuzaffar) [inline cmsBuild]: I think you need to drop `v` and add `z` for zip.

**Pull request closed without merging**